krux_boto/util.py

```diff
@@ -43,6 +43,25 @@ def get_instance_region():
         raise Error('get_instance_region failed to get the local instance region')
     return zone.rstrip(string.lowercase)
 
+def setup_hosts(hosts, accepted_domains=['krxd.net'], default='krxd.net'):
+    """
+    Loop through hosts to check if the domain matches any in accepted_domains. If not, append default.
+    This function will return a new list.
+
+    :param hosts: A list of hostname strings
+    :param accepted_domains: A list of accepted domain strings
+    :param default: A default domain name string to be appended to the hostnames
+
+    :return: A list of modified host name strings
+    """
+    new_hostnames = []
+    for i in range(len(hosts)):
+        # Python 2.6 support
+        if hosts[i][-8:len(hosts[i])] not in accepted_domains:
```

Comment:
Sorry that was hardcoded for 'krxd.net'! I fixed it to check for variable length domains.